recipes/zstd/all/conanfile.py

```diff
@@ -58,6 +61,12 @@ def _patch_sources(self):
             tools.replace_in_file(os.path.join(self._source_subfolder, "build", "cmake", "lib", "CMakeLists.txt"),
                                   "POSITION_INDEPENDENT_CODE On", "")
 
+        # to avoid generating mismatched cputype objects from assembler codes (especially cross compiling)
+        if tools.Version(self.version) >= "1.5.1" and self.settings.arch != "x86_64":
+            tools.replace_in_file(os.path.join(self._source_subfolder, "build", "cmake", "lib", "CMakeLists.txt"),
+                                  "file(GLOB DecompressSources ${LIBRARY_DIR}/decompress/*.c ${LIBRARY_DIR}/decompress/*.S)",
+                                  "file(GLOB DecompressSources ${LIBRARY_DIR}/decompress/*.c)")
+
```

Comment:
A patch in upstream CMakeLists based on `CMAKE_SYSTEM_PROCESSOR` would be more elegant, and could be submitted upstream.

Reply:
Thank you for your comments.
I am going to try to follow your idea.

Reply:
Fixed.